posts/2022/09-10-graphite-stacked-prs.md

```diff
@@ -0,0 +1,186 @@
+
+Because otherwise, we will miss things.
+
+### You will still need to rebase occasionally
+
+Graphite is good at rebasing, but there will still be conflicts. I feel like I am doing a fraction of the rebasing I would have done before (I still have nightmares), but there will still be rebasing occasionally.
+
+### Don't manage more than a stack or two
+
+The temptation is there to create many stacks or forks of stacks, but I recommend avoiding it because it brings back all the downsides of context switching. Instead, stay on the same task and add another PR to the stack.
+
+### Stick to one commit per branch
+
+When I first started, I tended to push multiple commits to the same branch and then use interactive rebase to either squash or drop commits. However, rebasing this way outside of Graphite managed to break Graphite's stack metadata, so I recommend using a single commit and amending it as you go.
+
+## My code workflow
+
+Some notes about tooling
+
+- `gh` is the [GitHub CLI](https://cli.github.com/)
+- `gt` is the [Graphite CLI](https://docs.graphite.dev/). Any commands passed in that Graphite does not recognize will fall through to git. So you do not need to use git commands anymore if you don't want to.
+
+### Create a ticket
+
+I like working with one ticket per PR, so thus start with a ticket that describes the changes and actions to be taken once merged.
+
+### Create a commit, branch, and WIP PR
+
+The long way of doing this would be
+
+```bash
+git add .
+git commit -m "COMMIT MESSAGE"
+gt branch create BRANCH_NAME
+gt branch submit
+```
+
+I created a fish alias to do this in one command.
+
+```bash
+# gb BRANCH_NAME COMMIT MESSAGE
+
+function gb
+  gt branch create "$argv[1]" -a -m "$argv[2..-1]" --no-interactive && \
+  gt branch submit --no-interactive
+end
+
+```
+
+### Amend commit
+
+Add onto your existing work by amending the current commit instead of pushing up a new one.
+
+```bash
+git add .
+git commit --amend --no-edit
+git push --force-with-lease
+```
+
+I recommend aliasing this too.
+
+### Set PR as ready for review
+
+Once I am ready for review, I run the following to turn my draft PR into a regular one.
+
+```bash
+gh pr ready
+```
+
+## Post-review care
+
+So we have a bunch of reviewed PRs; what's next? The nice thing is that we are not blocked, so that we can merge and deploy them once enough are ready to go (or we feel like deploying things).
+I periodically sync up remote and local and rebase off the main branch.
+
+```bash
+  gt repo sync --force
+  gt stack restack
+  gt stack submit --no-interactive
+```
+
+I generally merge branches one at a time into the main branch (starting from the "bottom" of a stack rather than the "top" if you visualize the stack as sitting on top of the default branch. Incrementally from the bottom allows me to do one deployment per PR and gradually push the work into production. Of course, while doing so, we read the ticket notes so we don't forget a migration or worse.
```

Comment:
Once again a mix between "i" and "we". And the second sentence of this paragraph is incomplete